### PR TITLE
enhance: adding a branch field to scheduled builds

### PR DESCRIPTION
**Pull request by claire1618**

### CODE REVIEW

**Review by ecrupper — CHANGES_REQUESTED** [review]

File: api/schedule/create.go [hunk 1]
```diff
@@ -136,6 +136,7 @@ func CreateSchedule(c *gin.Context) {
 	s.SetCreatedAt(time.Now().UTC().Unix())
 	s.SetUpdatedAt(time.Now().UTC().Unix())
 	s.SetUpdatedBy(u.GetName())
+	s.SetBranch(input.GetBranch())
```
[the remark this answers is not in the document]
Comment (claire1618) [reply api/schedule/create.go]: Good idea, I updated the code to reflect that

**Review by KellyMerrick — APPROVED** [review]
lgtm

File: api/schedule/update.go [hunk 2]
```diff
@@ -127,6 +127,7 @@ func UpdateSchedule(c *gin.Context) {
 
 	// set the updated by field using claims
 	s.SetUpdatedBy(u.GetName())
+	s.SetBranch(input.GetBranch())
```
Comment (ecrupper) [inline api/schedule/update.go:130]: We'll want to follow the same pattern as above in order to make sure an update payload like this: 
```json
{
    "entry": "0 0 * * *"
}
```
Doesn't result in the branch being set to an empty string.

```go
if input.GetBranch() != "" {
    s.SetBranch(input.GetBranch)
}
```

**Review by ecrupper — CHANGES_REQUESTED** [review]

**Review by ecrupper — APPROVED** [review]

File: api/schedule/update.go (hunk 2 above)
Comment (claire1618) [reply api/schedule/update.go]: I updated the code to reflect that

**Review by plyr4 — APPROVED** [review]

**Pull request merged**